tests/e2e/functional_test.go

```diff
@@ -30,12 +31,162 @@ type FunctionalSuite struct {
 	E2ESuite
 }
 
-// TODO
-func (s *FunctionalSuite) TestCreateGitSync() {
-	w := s.Given().GitSync("@testdata/gitsync.yaml").InitializeGitRepo().
+// GitSync testing for Numaflow with raw manifests
+func (s *FunctionalSuite) TestSimpleNumaflowGitSync() {
+
+	// create GitSync and initialize Git repo with initial manifest files
+	w := s.Given().GitSync("@testdata/gitsync.yaml").InitializeGitRepo("numaflow/initial-commit").
 		When().
 		CreateGitSyncAndWait()
 	defer w.DeleteGitSyncAndWait()
+
+	// verify Numaflow installation and pipeline creation
+	w.Expect().ResourcesExist("numaflow.numaproj.io/v1alpha1", "interstepbufferservices", []string{"default"})
+	w.Expect().ResourcesExist("numaflow.numaproj.io/v1alpha1", "pipelines", []string{"simple-pipeline"})
+
+	// verify that the CommitStatus of GitSync is currently synced with initial commit
+	w.Expect().CheckCommitStatus()
+
+	// pushing new pipeline file to repo
+	// wait currently necessary as controller takes time to reconcile and sync properly
+	w.PushToGitRepo("numaflow/modified", []string{"http-pipeline.yaml"}, false).Wait(30 * time.Second)
+
+	// verify that new pipeline has been created and GitSync is synced to newest commit
+	w.Expect().ResourcesExist("numaflow.numaproj.io/v1alpha1", "pipelines", []string{"http-pipeline"})
+	w.Expect().CheckCommitStatus()
+
+	// edit existing pipeline in repo to have additional vertex
+	w.PushToGitRepo("numaflow/modified", []string{"sample_pipeline.yaml"}, false).Wait(30 * time.Second)
+
+	// verify that spec of simple-pipeline was updated with new vertex
+	w.Expect().ResourcesExist("numaflow.numaproj.io/v1alpha1", "pipelines", []string{"simple-pipeline"})
+	// w.Expect().VerifyResourceSpec("numaflow.numaproj.io/v1alpha1", "pipelines", "simple-pipeline", "vertices", "")
+	w.Expect().CheckCommitStatus()
+
+	// edit existing pipeline back to initial state
+	w.PushToGitRepo("numaflow/initial-commit", []string{"sample_pipeline.yaml"}, false).Wait(30 * time.Second)
+
+	w.Expect().ResourcesExist("numaflow.numaproj.io/v1alpha1", "pipelines", []string{"simple-pipeline"})
+	w.Expect().ResourcesDontExist("numaflow.numaproj.io/v1alpha1", "vertices", []string{"simple-pipeline-another-out"})
+	w.Expect().ResourcesExist("numaflow.numaproj.io/v1alpha1", "vertices", []string{"simple-pipeline-in", "simple-pipeline-cat", "simple-pipeline-out"})
```

Comment:
I don't think you can do these tests involving vertices, right? this requires a Numaflow Controller to be running

Reply:
That's right, these can be removed in favor of checking the spec.